### ISSUE TITLE
Add support for emotion's reusable media queries (`${...}` inside a CSS template string)

### ISSUE THREAD

**Issue by kripod** [prettier/prettier#3380]
**Prettier 1.8.2**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuc0DOMAEAjATnAQwGsAHCAS1jUwF5NgAdKTTAelcwDkBXAWzlzkwBADaYAbqO5xqAd3Iix+NNxFZKmApl6UAtPIAmMABaYAjtNwBPJizS9RIpJgCsAdgBsAGlva4B8j5nTwAOH2ZMEQJcAHM4ZwBOBIAmcJYADwAZaLjnAEZkgAZCtLYOAGUYQSgYiSkZTHlFHDhMbjR-TWpyNF8YRwAFY2h4zAByAAoHdP1yI2NnAGYPQpJ0gEpNKANMKb1jOHIY4xhggBZVjbGmAF8AbiYmOHSyXCxIKAxzS3IGugB5bAAKzgYBgADoiHArGgJnhCKQKFR1uD8AZuGA4BMJgQwGA+KoCDAILgvJECNg4CJNjQAHz0XzkABmuxgVhIcAgLPhxDIlBgaAA2lFKSIALq0Gh0MYYaoxMabRgRFi4-G8QnE3DCilUiV0Cbgw05NA02m+FgsMBoNAAA3NFpYAAF+AEtBMACTAHmI-lCkW6m6K+0Olieq2ww3g43re7Bi03OM2h4RG6YKkdBnKzR4glRTXa0V63aR6O0M1Zy3Wu0Vi3O-zkN06KCzebOT3evlUAsBtZBmsOsPWg1G2Im2P9zAJmtJ3xT3z4GDcXDMVW5okk5M3MnAQPJpgfL4kaIEGK4AgkUx0cPVlhM6AwXRocgAL1GBTWe4inosAl+aHBLqBLwN4hre96Pi+b5nGsvg2nOX7AD+ggyOCUSxHAIFgbAEGvvkHgfrBU5JiAXggBAJAwOQ6DIKABAYMgTKiB0pF4LiUIwOUR5gJQMTIFU0ikQYEBgAxTFwKRlAdG8AxnjEDiiSIzEgECaDpAAQmeYDseUBD8JklBwApSkQNwMAkKZyRGeJIBHrgUnICA2AUlYIjQCRNnVDAADqcwmMgIQlB5EAdF5Z4kHxuACSA+AWOQ+AySe8lIIxinWR0OgRVFT41CIcAAIrcBA8BWaR-TYD58zIKkIBVAQCg8QAwhAvBJSAUAjO57RwAAKhSaBWTcNxAA)

**Input:**
```jsx
const breakpoints = {
  // Numerical values will result in a min-width query
  small: 576,
  medium: 768,
  large: 992,
  xLarge: 1200,
  // String values will be used as is
  tallPhone: '(max-width: 360px) and (min-height: 740px)'
};

export const queries = Object.keys(breakpoints).reduce((accumulator, label) => {
  if (typeof breakpoints[label] === 'string') {
    accumulator[label] = (...args) =>
      css`
        @media (${breakpoints[label]}) {
          ${css(...args)};
        }
      `;
  } else {
    accumulator[label] = (...args) =>
      css`
        @media (min-width: ${breakpoints[label]}px) {
          ${css(...args)};
        }
      `;
  }

  return accumulator;
}, {});

const paragraph = css`
  font-size: 12px;

  ${queries.medium`
        font-size: 14px
  `}

  ${queries.large`
    font-size: 16px;
  `}
`;
```

**Output:**
```jsx
const breakpoints = {
  // Numerical values will result in a min-width query
  small: 576,
  medium: 768,
  large: 992,
  xLarge: 1200,
  // String values will be used as is
  tallPhone: "(max-width: 360px) and (min-height: 740px)"
};

export const queries = Object.keys(breakpoints).reduce((accumulator, label) => {
  if (typeof breakpoints[label] === "string") {
    accumulator[label] = (...args) =>
      css`
        @media (${breakpoints[label]}) {
          ${css(...args)};
        }
      `;
  } else {
    accumulator[label] = (...args) =>
      css`
        @media (min-width: ${breakpoints[label]}px) {
          ${css(...args)};
        }
      `;
  }

  return accumulator;
}, {});

const paragraph = css`
  font-size: 12px;

  ${queries.medium`
        font-size: 14px
  `} ${queries.large`
    font-size: 16px;
  `};
`;

```

**Expected behavior:**
[Emotion's reusable media queries](https://emotion.sh/#reusable-media-queries) make Prettier "escape" from CSS mode. I think that CSS mode should be activated for every `${...}` descendant of a CSS template literal.

**Comment by duailibe** [comment]
Is there a way we can detect `queries.medium` is CSS in a standard way?

**Comment by suchipi** [comment]
Strictly speaking, it isn't safe to do. Although I can't think of real-world cases where it'd be a problem, parsing tagged template literals inside css tagged template literals assumes that the *input* of the tagged template literal inside the interpolation is CSS, which seems slightly riskier to assume than it would be to only assume the *output* is.

Maybe a better solution would be support for `prettier-use-parser` comments or something?
```js
css`
  ${/* prettier-use-parser css */queries.medium`color: "red";`}
`;
```

If emotion has a standard convention for what you should name your media queries object, detecting that inside of a CSS tagged template literal would be my first choice... but it doesn't seem that there's anything like that.

**Comment by lydell** [comment]
Is this the same as #3123?

**Comment by kripod** [comment]
@lydell Not really, because nested code (inside `${...}`) doesn’t get formatted at all.

**Comment by duailibe** [comment]
I think it's the same as #3123 and (a part of) #2300
